pkg/virt-controller/watch/vmi.go

```diff
@@ -310,11 +310,17 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			if isPodOwnedByHandler(pod) {
 				// vmi is still owned by the controller but pod is already handed over,
 				// so let's hand over the vmi too
-				vmiCopy.Status.Interfaces = []virtv1.VirtualMachineInstanceNetworkInterface{
-					{
-						IP: pod.Status.PodIP,
-					},
+
+				interfaces := make([]virtv1.VirtualMachineInstanceNetworkInterface, 0)
+				for _, network := range vmi.Spec.Networks {
+					ifc := virtv1.VirtualMachineInstanceNetworkInterface{Name: network.Name}
+					if network.NetworkSource.Pod != nil {
+						ifc.IP = pod.Status.PodIP
+					}
+					interfaces = append(interfaces, ifc)
```

Comment:
Can we move the append inside the `if`? I think that the controller can only add a status for pod interfaces ever. Most of the k8s objects normally only add something if they give additional value. I am not sure if we want a list consisting mostly of names alone. `virt-handler` can later on fill in any missing entries independent of the controller, if it has additional information (e.g. via the guest-agent). @mmirecki what do you think?

Reply:
Can you add a test to `vmi_test.go`?

Reply:
There is some value of having a complete list of interfaces in the status, even if we just display the names. It shows we know about the interface, but not about it's mac/ip (not assiged yet, which is also some kind of info).
But having no entry in status could also mean that the ifc has not been created yet, which I'm also fine with.
Moving to if.